src/main/java/org/apache/sling/commons/contentdetection/internal/ContentAwareMimeTypeServiceImpl.java

```diff
@@ -52,10 +52,16 @@ public String getMimeType(String filename, InputStream content) throws IOExcepti
         if(!content.markSupported()) {
             throw new IllegalArgumentException("Supplied InputStream does not support mark/reset");
         }
-        TikaInputStream stream = TikaInputStream.get(content);
-        Metadata metadata = new Metadata();
-        metadata.set(Metadata.RESOURCE_NAME_KEY, filename);
-        MediaType mediaType = detector.detect(stream, metadata);
+        MediaType mediaType;
+        TemporaryResources tmp = new TemporaryResources();
+        try {
```

Comment:
Please use https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html instead!

Reply:
ok, so I will put it like this
try (TemporaryResources tmp = new TemporaryResources()){ ...} 
Is it fine?